File: src/IsbInterop.CSharp/ServiceForms/Wrappers/ServiceFactory.cs
namespace IsbInterop.ServiceForms.Wrappers
{
  /// <summary>
  /// Обертка над IServiceFactory.
  /// </summary>
  public class ServiceFactory : IsbComObjectWrapper, IServiceFactory
  {
    /// <summary>
    /// Получает проводник системы.
    /// </summary>
    /// <param name="isMain">Признак получения главной формы проводника системы.</param>
    /// <returns>Проводник системы.</returns>
    public IEdmsExplorer GetExplorer(bool isMain)
    {
      var rcwExplorer = InvokeRcwInstanceMethod("GetExplorer", isMain);
      return new EdmsExplorer(rcwExplorer, Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIServiceFactory">COM-объект IServiceFactory.</param>
    /// <param name="scope">Область видимости.</param>
    internal ServiceFactory(object rcwIServiceFactory, IScope scope)
      : base(rcwIServiceFactory, scope) { }
  }
}

File: src/IsbInterop/ServiceForms/Wrappers/ServiceFactory.vb
Namespace ServiceForms.Wrappers

  ''' <summary>
  ''' Обертка над IServiceFactory.
  ''' </summary>
  Public Class ServiceFactory
    Inherits IsbComObjectWrapper
    Implements IServiceFactory

    ''' <summary>
    ''' Получает проводник системы.
    ''' </summary>
    ''' <param name="isMain">Признак получения главной формы проводника системы.</param>
    ''' <returns>Проводник системы.</returns>
    Public Function GetExplorer(isMain As Boolean) As IEdmsExplorer Implements IServiceFactory.GetExplorer
      Dim rcwExplorer = InvokeRcwInstanceMethod("GetExplorer", isMain)
      Return New EdmsExplorer(rcwExplorer, Scope)
    End Function

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIServiceFactory">COM-объект IServiceFactory.</param>
    ''' <param name="scope">Область видимости.</param>
    Friend Sub New(rcwIServiceFactory As Object, scope As IScope)
      MyBase.New(rcwIServiceFactory, scope)
    End Sub
  End Class
End Namespace